test/unit/repositories/test_image.py

```diff
@@ -84,3 +86,79 @@ def test_create(self):
         self.mock_create(IMAGE_IN_DATA_ALL_VALUES)
         self.call_create()
         self.check_create_success()
+
+
+class ListDSL(ImageRepoDSL):
+    """Base class for `list` tests."""
+
+    _expected_image_out: list[ImageOut]
+    _obtained_image_out: list[ImageOut]
```

Comment:
```suggestion
    _expected_image_out: list[ImageOut]
     _entity_id_filter: Optional[str]
     _parimary_filter: Optional[boolean]
     _obtained_image_out: list[ImageOut]
```
I would prefer this, while assigning them in the `call_list` and automatically using them in the `check_list_success`, as then the tests don't care about any implementation except the parameters the function is provided with. (IMS also does it this way)